File: gek-core/src/main/java/xyz/fslabo/common/codec/CodecException.java
package xyz.fslabo.common.codec;

/**
 * Exception for codec.
 *
 * @author fredsuvn
 */
public class CodecException extends RuntimeException {

    /**
     * Empty constructor.
     */
    public CodecException() {
    }

    /**
     * Constructs with exception message.
     *
     * @param message exception message
     */
    public CodecException(String message) {
        super(message);
    }

    /**
     * Constructs with exception message and exception cause.
     *
     * @param message exception message
     * @param cause   exception cause
     */
    public CodecException(String message, Throwable cause) {
        super(message, cause);
    }

    /**
     * Constructs with exception cause.
     *
     * @param cause exception cause
     */
    public CodecException(Throwable cause) {
        super(cause);
    }
}

File: boat-codec/src/main/kotlin/xyz/srclab/common/codec/CodecException.kt
package xyz.srclab.common.codec

import xyz.srclab.common.base.DEFAULT_SERIAL_VERSION
import java.io.Serializable

/**
 * Codec exception.
 */
open class CodecException @JvmOverloads constructor(
    name: String, cause: Throwable? = null
) : RuntimeException(name, cause), Serializable {
    companion object {
        private val serialVersionUID: Long = DEFAULT_SERIAL_VERSION
    }
}